File: repasoHerenciajava/Ayudante.java
import java.util.ArrayList;
class Ayudante extends Personal {
    String semestre;
    ArrayList<String> materiasDicta;
    public Ayudante(String name, String email, String password, String semestre) {
        super(name, email, password);
        this.semestre = semestre;
        this.materiasDicta = new ArrayList<String>();
    }
}

File: repasoHerenciajs/ayudante.js
import Personal from './personal.js'
class Ayudanate extends Personal {
    constructor(name, email, password, age, semestre) {
        super(name, email, password, age);
        this.materiasDicta = new Array();
        this.semestre = semestre;
    }
}

export default Ayudanate;